### PR TITLE
feat: surface ErrorInfo fields in google-cloud-errors

### PR DESCRIPTION
**Pull request by viacheslav-rostovtsev**

### CODE REVIEW

File: google-cloud-errors/lib/google/cloud/errors/version.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@
 module Google
   module Cloud
     module Errors
-      VERSION = "1.1.0".freeze
+      VERSION = "1.2.0".freeze
```
Comment (dazuma) [inline google-cloud-errors/lib/google/cloud/errors/version.rb:19]: Revert this change. Release-please will update the version.

File: google-cloud-errors/lib/google/cloud/errors.rb [hunk 2]
```diff
@@ -118,6 +118,59 @@ def status_details
...
+      # given that the following is true:
+      #   * `status_details` exists and is an array
+      #   * there is exactly one `::Google::Rpc::ErrorInfo` object in the `status_details` array
+      #
+      # @return [::Google::Rpc::ErrorInfo, nil]
+      def error_info
+        return nil unless status_details.is_a? Array
+        error_infos = status_details.find_all { |status| status.is_a?(::Google::Rpc::ErrorInfo) }
+        return nil unless error_infos.length == 1
+        error_infos[0]
```
Comment (dazuma) [inline google-cloud-errors/lib/google/cloud/errors.rb:132]: Consider memoizing this so the logic doesn't get rerun every time `error_info` is called. (Note that the other methods below already call `error_info` twice each.)

**Review by dazuma — CHANGES_REQUESTED** [review]

File: google-cloud-errors/lib/google/cloud/errors/version.rb (hunk 1 above)
Comment (viacheslav-rostovtsev) [reply google-cloud-errors/lib/google/cloud/errors/version.rb]: :+1:

File: google-cloud-errors/lib/google/cloud/errors.rb (hunk 2 above)
Comment (viacheslav-rostovtsev) [reply google-cloud-errors/lib/google/cloud/errors.rb]: :+1:

**Review by dazuma — APPROVED** [review]
